zkevm-circuits/src/tx_circuit.rs

```diff
@@ -3922,17 +3929,47 @@ impl<F: Field> SubCircuit<F> for TxCircuit<F> {
         // Since each call data byte at least takes one row in RLP circuit.
         // For L2 tx, each call data byte takes two row in RLP circuit.
         assert!(block.circuits_params.max_calldata < block.circuits_params.max_rlp_rows);
+
+        // Calculate blob capacity usage
+        let chunk_txbytes_len = block.txs.iter().map(|tx| {
+            if tx.is_chunk_l2_tx() {
+                tx.rlp_signed.len()
+            } else {
+                0
+            }
+        }).sum::<usize>();
+        let blob_usage: f32 = chunk_txbytes_len as f32 / CHUNK_TXBYTES_BLOB_LIMIT.clone() as f32;
+
+        // Calculate tx circuit dynamic section usage
         let sum_calldata_len = block.txs.iter().map(|tx| tx.call_data.len()).sum::<usize>();
-        let max_calldata = if block.circuits_params.max_calldata == 0 {
-            // dynamic max_calldata
-            sum_calldata_len
+        let sum_access_list_len = block.txs.iter().map(|tx| {
+            if tx.access_list.is_some() {
+                let access_list = tx.access_list.clone().unwrap().0;
+                access_list.len() + access_list.iter().map(|al| al.storage_keys.len()).sum::<usize>()
+            } else {
+                0usize
+            }
+        }).sum::<usize>();
+
+        // With the introduction of access list, the max_calldata circuit parameter now has to share capacity between calldata and access list rows
+        // TODO: The max_calldata parameter should be renamed later to max_dynamic
+        let max_dynamic_data = if block.circuits_params.max_calldata == 0 {
+            // input-specific max_dynamic
+            sum_calldata_len + sum_access_list_len
         } else {
             block.circuits_params.max_calldata
         };
```

Comment:
We can add `max_access_list_items` in `block.circuit_params` and set the max of `[calldata_usage, access_list_usage, chunk_txbytes_len]` to be `highest_usage`.